docs/plugins/introduction.mdx

```diff
@@ -28,22 +34,36 @@ These plugins can be applied at the following steps:
 
 ## Architecture
 
-<Thumbnail src="/img/plugins/architecture.jpg" alt="Engine architecture with plugins" width="1000px" />
-
-Engine plugins are HTTP servers that run alongside DDN and can be written in any language capable of running an HTTP
-server. They are configured in DDN using DDN metadata. The engine sends HTTP requests to the plugin at the specified
-execution step, the plugin processes the request, and then sends a response back to the engine, which continues
-execution based on the plugin's response.
+```mermaid
+---
+title: Engine Plugins Execution Pipeline
+---
+graph RL
+    Client[Client] -->|Request| Authentication
+    Authentication --> Query_Parsing_and_Planning["Query parsing and planning"]
+    Query_Parsing_and_Planning --> Execute_Query["Execute query"]
+    Execute_Query <-->|Fetch Data| Data_Connector["Data connector"]
+    Execute_Query --> Post_Processing["Post-processing"]
+    Post_Processing --> Pre_Response_Hooks["Pre-response hooks"]
+    Pre_Response_Hooks -->|Response| Client
```

Comment:
```suggestion
    Post_Processing -->|Response| Client
```

Reply:
When will this name change come through on the engine, @paritosh-08? If it won't be before this week, we should probably leave this as is and then make the change in #841?

Reply:
Sorry, I should have been more clear. The pre-response hook doesn't send the response to the client. The respinse is sent to the client at the post-processing stage.

Reply:
The architecture diagram already have the post-processing node. I have just fixed the edge in the node.